### PR TITLE
feat: core uptake 6-1-22 

### PR DESCRIPTION
**Pull request by emilyjablonski**
## Issue

- Closes #issue
- Addresses #issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Please include a summary of the change and which issue(s) is addressed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## How Can This Be Tested/Reviewed?

Provide instructions so we can review.

Describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (emilyjablonski) [inline .github/workflows/codeql.yml:1]: Not sure if we want to bring this over, but seems like it wouldn't hurt?

File: .github/workflows/pre-release_components.yml [hunk 2]
```diff
@@ -17,13 +17,13 @@ jobs:
         run: git config --global user.email "github-actions@github.com"
 
       - name: Check out git repo
-        uses: actions/checkout@v3
```
Comment (emilyjablonski) [inline .github/workflows/pre-release_components.yml:20]: This is a downgrade, but this is what we have in core, and we don't / won't run this over here.

File: backend/core/src/auth/auth.module.ts [hunk 3]
```diff
@@ -46,7 +45,7 @@ import { UserPreferences } from "./entities/user-preferences.entity"
       }),
       inject: [ConfigService],
     }),
-    TypeOrmModule.forFeature([RevokedToken, User, Application, UserPreferences]),
```
Comment (emilyjablonski) [inline backend/core/src/auth/auth.module.ts:49]: Not sure here which to keep

File: backend/core/src/auth/dto/user-filter-type-to-field-map.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { UserFilterKeys } from "../types/user-filter-keys"
 
 export const userFilterTypeToFieldMap: Record<keyof typeof UserFilterKeys, string> = {
-  isPartner: "user_roles.isPartner",
```
Comment (emilyjablonski) [inline backend/core/src/auth/dto/user-filter-type-to-field-map.ts:4]: Not sure about this one

File: backend/core/src/auth/services/password.service.ts [hunk 5]
```diff
@@ -12,9 +12,8 @@ export class PasswordService {
   // passwordHash is a hidden field - we need to build a query to get it directly
   public async getUserWithPassword(user: User) {
     return await this.userRepository
-      .createQueryBuilder()
```
Comment (emilyjablonski) [inline backend/core/src/auth/services/password.service.ts:15]: Not sure about this one

File: backend/core/src/listings/listings.service.ts [hunk 6]
```diff
@@ -39,6 +46,42 @@ export class ListingsService {
     return getView(this.listingRepository.createQueryBuilder("listings"), "full").getViewQb()
   }
 
+  private static getOrderByCondition(
```
Comment (emilyjablonski) [inline backend/core/src/listings/listings.service.ts:49]: Not sure about this one

File: backend/core/src/listings/types/listings-api-extra-models.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { Expose } from "class-transformer"
+import { ApiProperty } from "@nestjs/swagger"
+import { OrderByFieldsEnum } from "./listing-orderby-enum"
+import { OrderParam } from "../../applications/types/order-param"
+
+export class ListingsApiExtraModels {
```
Comment (emilyjablonski) [inline backend/core/src/listings/types/listings-api-extra-models.ts:6]: Not sure about this one

File: backend/proxy/Dockerfile [hunk 8]
```diff
@@ -32,13 +32,16 @@ RUN NGINX_VERSION=`nginx -V 2>&1 | grep "nginx version" | awk -F/ '{ print $2}'`
...
 # configure and build
 RUN cd /tmp/nginx && \
     BASE_CONFIGURE_ARGS=`nginx -V 2>&1 | grep "configure arguments" | cut -d " " -f 3-` && \
     /bin/sh -c "./configure ${BASE_CONFIGURE_ARGS} --add-module=/tmp/ngx_cache_purge" && \
     make && make install && \
-    rm -rf /tmp/nginx*
+    rm -rf /tmp/nginx* \
 
-CMD /bin/bash -c "envsubst '\$PORT,\$BACKEND_HOSTNAME' < /etc/nginx/conf.d/default.conf.template > /etc/nginx/conf.d/default.conf" && nginx -g 'daemon off;'
+ENV PROTOCOL=https
```
Comment (emilyjablonski) [inline backend/proxy/Dockerfile:45]: Not sure about this

File: backend/proxy/default.conf [hunk 9]
```diff
@@ -19,44 +19,26 @@ server {
   proxy_cache_valid 500 0s;
 
   location ~* "\/listings\/[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}" {
-    proxy_cache_min_uses 1;
```
Comment (emilyjablonski) [inline backend/proxy/default.conf:22]: Not sure about this

File: sites/partners/src/listings/PaperListingForm/sections/AdditionalEligibility.tsx [hunk 10]
```diff
@@ -1,13 +1,32 @@
-import React from "react"
+import React, { useEffect } from "react"
 import { useFormContext } from "react-hook-form"
 import { t, GridSection, Textarea } from "@bloom-housing/ui-components"
 import { fieldMessage } from "../../../../lib/helpers"
 
-const AdditionalEligibility = () => {
+type AdditionalEligibilityProps = {
```
Comment (emilyjablonski) [inline sites/partners/src/listings/PaperListingForm/sections/AdditionalEligibility.tsx:6]: Not sure about this

File: backend/core/src/auth/auth.module.ts [hunk 11]
```diff
@@ -24,6 +23,7 @@ import { SmsMfaService } from "./services/sms-mfa.service"
 import { UserPreferencesController } from "./controllers/user-preferences.controller"
 import { UserPreferencesService } from "./services/user-preferences.services"
 import { UserPreferences } from "./entities/user-preferences.entity"
+import { TwilioModule } from "nestjs-twilio"
```
Comment (emilyjablonski) [inline backend/core/src/auth/auth.module.ts:26]: Note to self: use Detroit order

File: backend/core/src/auth/dto/user-filter-type-to-field-map.ts (hunk 4 above)
Comment (emilyjablonski) [inline backend/core/src/auth/dto/user-filter-type-to-field-map.ts:4]: Note to self: investigate

File: backend/core/src/auth/guards/local-auth.guard.ts [hunk 12]
```diff
@@ -1,5 +0,0 @@
-import { Injectable } from "@nestjs/common"
```
Comment (emilyjablonski) [inline backend/core/src/auth/guards/local-auth.guard.ts:1]: Note to self: Check if MFA is wanted on Detroit and if not ensuring it is set as false on user create

File: sites/partners/lib/hooks.ts [hunk 13]
```diff
@@ -45,42 +50,41 @@ export function useSingleListingData(listingId: string) {
...
+    view: "base",
+  }
+
+  if (sort) {
+    Object.assign(params, {
+      orderBy: sort?.filter((item) => item.orderBy).map((item) => item.orderBy),
+    })
+    Object.assign(params, {
+      orderDir: sort?.filter((item) => item.orderDir).map((item) => item.orderDir),
+    })
```
Comment (seanmalbert) [inline sites/partners/lib/hooks.ts:66]: @emilyjablonski 
If we're not yet brining over the listings order updates in the backend, then we can't yet add this either.

File: sites/partners/pages/index.tsx [hunk 14]
```diff
@@ -1,4 +1,4 @@
-import React, { useMemo, useContext, useEffect, useRef, useState, useCallback } from "react"
+import React, { useMemo, useContext, useState, useCallback, useEffect } from "react"
```
Comment (emilyjablonski) [inline sites/partners/pages/index.tsx:1]: test w sorting / searching removals

File: sites/partners/pages/users/terms.tsx [hunk 15]
```diff
@@ -0,0 +1,13 @@
+import React from "react"
+import FormsLayout from "../../layouts/forms"
+import { FormTerms } from "@bloom-housing/ui-components"
+
+const TermsPage = () => {
```
Comment (emilyjablonski) [inline sites/partners/pages/users/terms.tsx:5]: double check this flow

File: sites/partners/src/listings/PaperListingForm/UnitForm.tsx [hunk 16]
```diff
@@ -440,7 +467,16 @@ const UnitForm = ({ onSubmit, onClose, defaultUnit, nextId, draft }: UnitFormPro
                 labelClassName="sr-only"
                 register={register}
                 controlClassName="control"
-                options={numberOptions(10)}
+                options={numberOptions(numberOccupancyOptions)}
+                error={fieldHasError(errors?.minOccupancy)}
+                errorMessage={t("errors.minGreaterThanMaxOccupancyError")}
+                validation={{ max: maxOccupancy || numberOccupancyOptions }}
+                inputProps={{
+                  onChange: () => {
```
Comment (emilyjablonski) [inline sites/partners/src/listings/PaperListingForm/UnitForm.tsx:475]: make an issue to add to unit group form

File: sites/partners/src/listings/PaperListingForm/sections/ListingPhoto.tsx [hunk 17]
```diff
@@ -178,6 +187,7 @@ const ListingPhoto = () => {
           </GridCell>
         </GridSection>
       </GridSection>
+      <p className="field-sub-note">{t("listings.requiredToPublish")}</p>
```
Comment (emilyjablonski) [inline sites/partners/src/listings/PaperListingForm/sections/ListingPhoto.tsx:190]: remove

File: sites/partners/src/listings/PaperListingForm/UnitForm.tsx (hunk 16 above)
Comment (emilyjablonski) [reply sites/partners/src/listings/PaperListingForm/UnitForm.tsx:475]: already exists 👍

**Comment by seanmalbert** [comment]
closing for https://github.com/CityOfDetroit/bloom/pull/1357, which is the same as this

**Pull request closed without merging**